### DISCUSSION TITLE
How to get ACSelect value as string?

### DISCUSSION THREAD

**Discussion by fruktiz32** [Hieromon/AutoConnect#416] [How can I?] [answered]
Hello, thanks for your great library. I can not get selected option as an String. I'm using this and then trying to print as curr.value, but it _says invalid use of non-static member function_

  
`AutoConnectSelect& curr = coin->getElement<AutoConnectSelect>("currency");`
`Serial.println(curr.value);`

**Comment by Hieromon** [comment, marked as answer]
@fruktiz32 Hello, Thank you for your interest in AutoConnect.
ACSelect retains the provided values as choice strings by a slightly different implementation than other AutoConnectElements. It saves the choice strings as an array with [`std::vector<String>`](https://en.cppreference.com/w/cpp/container/vector), so the sketch cannot directly access the `value` member to get a determined string.
Use the [`value()`](https://hieromon.github.io/AutoConnect/apielements.html#value_6) function to get the string selected in ACSelect. So change your code to:
```cpp
Serial.println(curr.value());
```

**Reply by fruktiz32** [reply]
Thanks! That worked great!
